presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDictionarySelectiveStreamReader.java

```diff
@@ -201,7 +204,7 @@ private int readWithFilter(int[] positions, int positionCount)
                     int id = (int) value;
                     value = dictionary[id];
 
-                    if (!nonDeterministicFilter) {
+                    if (dictionaryFilterStatus != null) {
```

Comment:
What is the meaning of this flag?

Reply:
I think I figured it out.

Reply:
Say there is a filter on a column and the filter is deterministic. Say a > 10 && a < 20 && a%3 == 0 && a%5 ==0 .
The filter can be evaluated on each row. But if the column is dictionary encoded, the filter can be evaluated once for each dictionary index and cache it.

In this case, FlatMap dictionary results were about to be cached, which resulted in large memory consumption. 19GB. With this change, if a column is dictionary encoded inside flat maps, the results will not be cached.